src/CodeFlowDecorations.ts

```diff
@@ -284,33 +292,28 @@ export class CodeFlowDecorations implements Disposable {
      * @param sarifLocation raw sarif location used if location isn't mapped to get the user to try to map
      */
     private async updateSelectionHighlight(location: Location, sarifLocation?: sarif.Location): Promise<void> {
-        if (!this.activeDiagnostic) {
+        if (!this.activeDiagnostic || !this.activeDiagnostic.location.hasBeenMapped) {
             return;
         }
 
-        const runInfo: RunInfo = this.activeDiagnostic.runInfo;
-
-        const remappedLocation: Location | undefined = await LocationFactory.getOrRemap(
-            this.fileMapper,
-            runInfo,
-            location,
-            sarifLocation);
-
-        if (remappedLocation && remappedLocation.mapped && remappedLocation.uri) {
-            let locRange: Range | undefined = remappedLocation.range;
-            if (!locRange) {
-                return;
-            }
+        const mappedUri: Uri | undefined = this.activeDiagnostic.location.uri;
+        if (!mappedUri) {
+            return;
+        }
 
-            if (remappedLocation.endOfLine) {
-                locRange = new Range(locRange.start, new Position(locRange.end.line - 1, Number.MAX_VALUE));
-            }
+        let locRange: Range | undefined = location.range;
+        if (!locRange) {
+            return;
+        }
 
-            const textDocument: TextDocument = await  workspace.openTextDocument(remappedLocation.uri);
-            const textEditor: TextEditor = await   window.showTextDocument(textDocument, ViewColumn.One, true);
-            textEditor.setDecorations(CodeFlowDecorations.SelectionDecorationType, [{ range: locRange }]);
-            textEditor.revealRange(locRange, TextEditorRevealType.InCenterIfOutsideViewport);
+        if (location.endOfLine) {
+            locRange = new Range(locRange.start, new Position(locRange.end.line - 1, Number.MAX_VALUE));
         }
+
+        const textDocument: TextDocument = await workspace.openTextDocument(mappedUri);
+        const textEditor: TextEditor = await window.showTextDocument(textDocument, ViewColumn.One, true);
```

Comment:
Can this fail if someone opens and quickly closes a window?

Reply:
Yes, sort of. VSCode takes this as an "advisement" not a directive. The changes I made to "not put focus" back on the web-view actually addresses this concern as I also hit that issue during testing.

---
In reply to: [405160051](https://github.com/microsoft/sarif-vscode-extension/pull/218#discussion_r405160051) [](ancestors = 405160051)